### PR TITLE
fix some text field focus bugs

### PR DESCRIPTION
**Pull request by MicrocontrollersDev**
## Description

code analysis in mojira issue reports

## Related Issue(s)

Fixes [MC-151412](https://bugs.mojang.com/browse/MC-151412)
Fixes [MC-233042](https://bugs.mojang.com/browse/MC-233042)

### CODE REVIEW

File: common/src/main/java/cc/woverflow/debugify/mixins/client/mc233042/DirectConnectScreenMixin.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+@Mixin(DirectConnectScreen.class)
+public class DirectConnectScreenMixin {
+    @Shadow private TextFieldWidget addressField;
+
+    @Redirect(method = "init", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/widget/TextFieldWidget;setTextFieldFocused(Z)V"))
+    private void unfocusTextField(TextFieldWidget instance, boolean focused) {
+        this.addressField.setTextFieldFocused(false);
```
[the remark this answers is not in the document]
Comment (MicrocontrollersDev) [reply common/src/main/java/cc/woverflow/debugify/mixins/client/mc233042/DirectConnectScreenMixin.java:16]: it already is set to true, which is apparently what breaks the focus in the first place. setting it to false fixes it

File: common/src/main/java/cc/woverflow/debugify/mixins/client/mc151412/AddServerScreenMixin.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    protected AddServerScreenMixin(Text title) {
+        super(title);
+    }
+
+    @Shadow
+    private TextFieldWidget serverNameField;
+
+    @Inject(method = "init", at = @At("RETURN"))
+    private void setFocus(CallbackInfo ci) {
+        this.serverNameField.setTextFieldFocused(false);
```
[the remark this answers is not in the document]
Comment (MicrocontrollersDev) [reply common/src/main/java/cc/woverflow/debugify/mixins/client/mc151412/AddServerScreenMixin.java:24]: same here

File: common/src/main/java/cc/woverflow/debugify/mixins/client/mc233042/DirectConnectScreenMixin.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (MicrocontrollersDev) [reply common/src/main/java/cc/woverflow/debugify/mixins/client/mc233042/DirectConnectScreenMixin.java:16]: sure

**Comment by MicrocontrollersDev** [comment]
it seems that using just allowing the ide to add the mixin to the json automatically rearranges the order of the mixins so its no longer in numerical order. i don't think its worth keeping it in an order in that case so ill just fix the duplicate from the merge and leave it as is

File: common/src/main/resources/debugify-common.mixins.json [hunk 3]
```diff
@@ -18,7 +18,6 @@
     "client.mc165595.GuardianEntityRendererMixin",
     "client.mc231097.ClientPlayerEntityMixin",
     "client.mc233042.DirectConnectScreenMixin",
-    "client.mc234898.RealmsMainScreenMixin"
```
Comment (isXander) [inline common/src/main/resources/debugify-common.mixins.json]: Looks like you deleted a mixin!

Comment (MicrocontrollersDev) [reply common/src/main/resources/debugify-common.mixins.json]: yes https://github.com/W-OVERFLOW/Debugify/commit/94c656a4cdfdcb7938fe8c3fd5141a3858ffaeff added a duplicate which i removed. looking at the final changes, no mixin was removed

**Pull request merged**